tests/test_pylock.py

```diff
@@ -0,0 +1,472 @@
+    environment = 'release-pypi'
+    kind = 'GitHub'
+    repository = 'python-attrs/attrs'
+    workflow = 'pypi-package.yml'
+
+    [[packages]]
+    name = 'cattrs'
+    version = '24.1.2'
+    requires-python = '>=3.8'
+    dependencies = [
+        {name = 'attrs'},
+    ]
+    index = 'https://pypi.org/simple'
+    wheels = [
+    {name = 'cattrs-24.1.2-py3-none-any.whl', upload-time = 2024-09-22T14:58:34.812643+00:00, url = 'https://files.pythonhosted.org/packages/c8/d5/867e75361fc45f6de75fe277dd085627a9db5ebb511a87f27dc1396b5351/cattrs-24.1.2-py3-none-any.whl', size = 66446, hashes = {sha256 = '67c7495b760168d931a10233f979b28dc04daf853b30752246f4f8471c6d68d0'}},
+    ]
+
+    [[packages]]
+    name = 'numpy'
+    version = '2.2.3'
+    requires-python = '>=3.10'
+    wheels = [
+    {name = 'numpy-2.2.3-cp312-cp312-win_amd64.whl', upload-time = 2025-02-13T16:51:21.821880+00:00, url = 'https://files.pythonhosted.org/packages/42/6e/55580a538116d16ae7c9aa17d4edd56e83f42126cb1dfe7a684da7925d2c/numpy-2.2.3-cp312-cp312-win_amd64.whl', size = 12626357, hashes = {sha256 = '83807d445817326b4bcdaaaf8e8e9f1753da04341eceec705c001ff342002e5d'}},
+    {name = 'numpy-2.2.3-cp312-cp312-manylinux_2_17_x86_64.manylinux2014_x86_64.whl', upload-time = 2025-02-13T16:50:00.079662+00:00, url = 'https://files.pythonhosted.org/packages/39/04/78d2e7402fb479d893953fb78fa7045f7deb635ec095b6b4f0260223091a/numpy-2.2.3-cp312-cp312-manylinux_2_17_x86_64.manylinux2014_x86_64.whl', size = 16116679, hashes = {sha256 = '3b787adbf04b0db1967798dba8da1af07e387908ed1553a0d6e74c084d1ceafe'}},
+    ]
+
+    [tool.mousebender]
+    command = ['.', 'lock', '--platform', 'cpython3.12-windows-x64', '--platform', 'cpython3.12-manylinux2014-x64', 'cattrs', 'numpy']
+    run-on = 2025-03-06T12:28:57.760769
+    """  # noqa: E501
+)
+
+
+def test_toml_roundtrip() -> None:
+    pylock_dict = tomllib.loads(PEP751_EXAMPLE)
+    pylock = Pylock.from_dict(pylock_dict)
+    # Check that the roundrip via Pylock dataclasses produces the same toml
+    # output, modulo TOML serialization differences.
+    assert tomli_w.dumps(pylock.to_dict()) == tomli_w.dumps(pylock_dict)
+
+
+@pytest.mark.parametrize("version", ["1.0", "1.1"])
+def test_pylock_version(version: str) -> None:
+    data = {
+        "lock-version": version,
+        "created-by": "pip",
+        "packages": [],
+    }
+    Pylock.from_dict(data)
+
+
+def test_pylock_unsupported_version() -> None:
+    data = {
+        "lock-version": "2.0",
+        "created-by": "pip",
+        "packages": [],
+    }
+    with pytest.raises(PylockUnsupportedVersionError):
+        Pylock.from_dict(data)
+
+
+def test_pylock_invalid_version() -> None:
+    data = {
+        "lock-version": "2.x",
+        "created-by": "pip",
+        "packages": [],
+    }
+    with pytest.raises(PylockValidationError) as exc_info:
+        Pylock.from_dict(data)
+    assert str(exc_info.value) == "Invalid version: '2.x' in 'lock-version'"
```

Comment:
Probably better to just test that "2.x" is somewhere in the message.

Same goes for any of the other tests that's doing an exact exception message comparison instead of just testing the key details are somewhere in the string.